scrc/dataset_creation/dataset_creator.py

```diff
@@ -356,13 +359,17 @@ def load_decision(self, court_string, engine):
         table = 'decision d LEFT JOIN language ON language.language_id = d.language_id'
         columns = 'd.*, extract(year from d.date) as year, language.iso_code as lang'
         where = f"d.decision_id IN {where_string_court('decision_id', court_string)}"
-        return next(self.select(engine, table, columns, where, chunksize=self.get_chunksize()))
+        return next(self.select(engine, table, columns, where, chunksize=self.get_chunksize()), pd.DataFrame())
 
     def load_file_number(self, decision_ids, df, engine):
         self.logger.info('Loading File Number')
         table = f"{join_tables_on_decision(['file_number'])}"
         where = f"file_number.decision_id IN ({','.join(decision_ids)})"
-        file_number_df = next(self.select(engine, table, "file_numbers", where, None, self.get_chunksize()))
+        try:
+            file_number_df = next(self.select(engine, table, "file_numbers", where, None, self.get_chunksize()))
+        except StopIteration:
+            self.logger.info("Exception caught in line 352: load_file_number")
+            return df
```

Comment:
why do you use the StopIteration here instead of the default vaue in the next function as in the other places?
The line number is hard-coded. This may be very confusing as soon as it is not correct anymore.

Reply:
I think I used this because we have an access to the `file_number_df` and a return statement below, but we could probably just use a default value and then an if-statement to check if the df is empty.
Is there a preference for the second option?

The line number was just a temporary arrangement which I've forgot to change.

Reply:
I don't have a preference, but I think it makes sense to do the same thing in the same situations. And I think the default value in the next call is elegant.

Reply:
Yeah, true, changed it.